### DISCUSSION TITLE
Trying to understand when an async atom isn't updating when dependencies change.

### DISCUSSION THREAD

**Discussion by eikonomega** [pmndrs/jotai#1891] [Q&A]
Hello friends,

Ok, I've got this derived atom (which I've simplified, but shown all the relevant parts) that inconsistently updates and I can't figure out why.

```typescript
/** Retrieve all analytics events for the current project. */
export const allEventsAtom = atom(async (get) => {
  const client = get(supabaseClientAtom);
  const dateRange = get(selectedDateRangeAtom);
  const [filteredExperiences, currentProject] = await Promise.all([
    get(experiencesFilteredByEnclosingURLAtom),
    get(currentAtlasProjectAtom),
  ]);

  console.log("AllEventsAtom", filteredExperiences.length);
  
  // Just returning an empty array for this example.
  return []
}
  ```
  
  For some reason, this atom doesn't update each time the dependencies change, particularly `experiencesFilteredByEnclosingURLAtom` which is another async atom. It does update sometimes, but not others.
  
  I have verified that `experiencesFilteredByEnclosingURLAtom` often updates without this atom also updating. Does anyone have any ideas as to why this might be?

Thanks!

**Comment by t6adev** [comment]
I am just curious about this problem.
There is no issue with the example.
https://codesandbox.io/s/async-atoms-in-read-sample-63yd7k?file=/src/App.tsx
